docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,155 @@
+cdouble if double else float.
+
+The following factory functions can be used to create complex tensors:
+
+- :func:`torch.tensor`
+- :func:`torch.as_tensor`
+- :func:`torch.empty`
+- :func:`torch.rand`
+- :func:`torch.randn`
+- :func:`torch.as_strided`
+- :func:`torch.from_numpy`
+- :func:`torch.zeros`
+- :func:`torch.ones`
+- :func:`torch.full`
+- :func:`torch.eye`
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real`:
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+
+Angle and abs
+-------------
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Serialization
+-------------
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+Autograd Support (with and without view_as_real, view_as_complex)
+-----------------------------------------------------------------
+
+
+::
+
+     x = torch.randn(2, 2, 2, dtype=torch.double, requires_grad=True)>>> x = torch.randn(2, 2, 2, dtype=torch.double, requires_grad=True)
+     >>> y = x.detach().requires_grad_(True)
+     >>> x0 = x.clone()
+     >>> x1 = torch.view_as_complex(x0)
+     >>> x2 = torch.view_as_real(x1)
+     >>> x2.mul_(2)
+     tensor([[[ 4.2425, -0.1076],
+          [ 3.2731,  2.3156]],
+
+          [[ 4.1179,  0.7358],
+          [-1.7711, -0.4389]]], dtype=torch.float64,
+          grad_fn=<ViewAsRealBackward>)
+     >>> x2.sum().backward()
+     >>> y0 = y.clone()
+     >>> y0.mul_(2)
+     tensor([[[ 4.2425, -0.1076],
+          [ 3.2731,  2.3156]],
+
+          [[ 4.1179,  0.7358],
+          [-1.7711, -0.4389]]], dtype=torch.float64, grad_fn=<MulBackward0>)
+     >>> y0.sum().backward()
+     >>> x.grad
+     tensor([[[2., 2.],
+          [2., 2.]],
+
+          [[2., 2.],
+          [2., 2.]]], dtype=torch.float64)
+     >>> y.grad
+     tensor([[[2., 2.],
+          [2., 2.]],
+
+          [[2., 2.],
+          [2., 2.]]], dtype=torch.float64)
+
+We also do not support the following subsystems:
+
+Quantization
+
+JIT
+distributions
+
+multiprocessing
+
+distributed
```

Comment:
If distributed doesn't work, it doesn't for any good reason, and we should fix it (shouldn't be hard)

Reply:
Synced with @mrshenli and we will have to add mappings between complex and float dtypes for data transfer (because for eg. [nccl](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/types.html?highlight=ncclfloat) doesn't support complex dtypes) and add some tests to ensure that it works correctly. These changes will be included as part of `1.7` because it's too last minute to add it in `1.6`.